tests/unit/request/test_base.py

```diff
@@ -57,3 +74,55 @@ def test_request_generator_repr():
     assert repr(generator) == (
         "RequestGenerator(mode=sync, async_queue_size=100, tokenizer=None)"
     )
+
+
+@pytest.mark.regression
+def test_request_generator_create_item_not_implemented():
+    with pytest.raises(TypeError):
+        class IncompleteRequestGenerator(RequestGenerator):
+            pass
+
+        IncompleteRequestGenerator()
+
+    class IncompleteCreateItemGenerator(RequestGenerator):
+        def create_item(self):
+            super().create_item()
+
+    generator = IncompleteCreateItemGenerator()
+    with pytest.raises(NotImplementedError):
+        generator.create_item()
+
+
+@pytest.mark.regression
+def test_request_generator_iter_calls_create_item():
+    generator = TestRequestGenerator(mode="sync")
+    generator.create_item = Mock(
+        return_value=TextGenerationRequest(prompt="Mock prompt")
+    )
+
+    items = []
+    for item in generator:
+        items.append(item)
+        if len(items) == 5:
+            break
+
+    assert len(items) == 5
+    generator.create_item.assert_called()
+
+
+@pytest.mark.regression
+def test_request_generator_async_iter_calls_create_item():
+    generator = TestRequestGenerator(mode="sync")
+    generator.create_item = Mock(
+        return_value=TextGenerationRequest(prompt="Mock prompt")
+    )
+
+    items = []
+    for item in generator:
+        items.append(item)
+        if len(items) == 5:
+            break
+
+    generator.stop()
+    assert len(items) == 5
+    generator.create_item.assert_called()
```

Comment:
Just out of curiosity. Do we need to test the length of the `items` list?
The statement: `if len(items) == 5` already defines that, kinda.

_tldr: the code in the test function itself already guaranty that the length will be 5 and we don't have to test it more_

I feel that it could be checked if the length of the returned items is mocked somewhere inside the generator. But this assertion looks useless to me.

What do you think?

Reply:
good call on this, this was supposed to be checking that the internal state did not change after calling generaator.stop rather than the items generated. I'll git up a fix shortly